### PR TITLE
Fix  #1219: Emoticon not showing up after being sent

### LINKED ISSUE #1219: Emoticon not showing up after being sent
### Problem
In the file frontend/views/containers/chatroom/MessageReactions.vue, we can find the codes to render emoticons from the emoticonsList.

This emoticonsList is a prop which is sent from the frontend/views/containers/chatroom/ChatMain.vue thru the MessageBase.vue component.

The problem is that emoticonsList is updated every time user send/remove emoticons but UI doesn't keep updated.
Emoticon Updates

They are all MESSAGE_ACTION_TYPES.EDIT_MESSAGE event. And it handles well. But UI doesn't update...
image

### Solution

Find why the UI is not being updated and force to refresh

### PR DESCRIPTION
**Pull request by pieer**

### CODE REVIEW

File: frontend/views/containers/chatroom/ChatMain.vue [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
         component(
           :is='messageType(message)'
-          :key='message.id'
+          :key='message.id + Math.random()'
```
Comment (taoeffect) [inline frontend/views/containers/chatroom/ChatMain.vue:31]: This change is related to the "duplicate keys" bug of #1227, but a more correct fix should be used. /cc @Silver-IT

Comment (Silver-IT) [reply frontend/views/containers/chatroom/ChatMain.vue:31]: Make sense.

File: frontend/views/containers/chatroom/ChatMain.vue [hunk 2]
```diff
@@ -275,7 +275,7 @@ export default ({
         if (alreadyAdded >= 0) {
           emoticons[emoticon].splice(alreadyAdded, 1)
           if (emoticons[emoticon].length === 0) {
-            delete this.messages[emoticon]
+            delete this.messages[index].emoticons[emoticon]
```
Comment (Silver-IT) [inline frontend/views/containers/chatroom/ChatMain.vue:278]: @pieer, I think you need to make a new branch from `chat-contract-messaging` branch and make pull request to `chat-contract-messaging` branch. Actually, making reactions is not implemented in the `main` branch.
I think this is my mistake that I didn't tell you before. You couldn't know.

**Review by Silver-IT — COMMENTED** [review]
Nice work, @pieer. I think you worked in the wrong place.
You need to create a new branch from `chat-contract-messaging` brach and also make a PR to `chat-contract-messaging` branch. Making reactions are not implemented in `main` branch and it is implemented in `chat-contract-messaging` branch and the issue was related to the `chat-contract-messaging` branch. I was offline when you start working on this. I should tell you before. That's my mistake. Sorry for that.

**Comment by pieer** [comment]
Better fix implemented chat-contract-messaging branch

**Pull request closed without merging**